src/lower/lowerer_impl_imperative.cpp

```diff
@@ -2067,13 +2079,27 @@ Stmt LowererImplImperative::lowerForallBody(Expr coordinate, IndexStmt stmt,
   // Code to append coordinates
   Stmt appendCoords = appendCoordinate(appenders, coordinate);
 
+  std::vector<Stmt> stmts;
+  // BEGIN GJ
+  if (caseLattice.iterators().size() > 1) {
+    // YOLO
+    // for (auto it : caseLattice.iterators()) {
+    for (auto it : getModeIterators(combine(caseLattice.iterators(), caseLattice.locators()))) {
+      Expr ivar = it.getIteratorVar();
+      stmts.push_back(compoundAssign(ivar, 1));
```

Comment:
When a merge point succeeds, we need to increment all iterators by 1, otherwise galloping won't move.

Reply:
Why do you also need to increment the locators?

Reply:
I do not, was just blinding copying code :)